### PR TITLE
{lib}[intel/2016a] The Virtual Brain (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/bbb06576303d50420eed1289a09d258b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node2722.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/24da9960fda73435fef10de130c4b9e9 for a full test report.

File: easybuild/easyconfigs/p/Pillow/Pillow-3.2.0-intel-2016a-Python-2.7.11-freetype-2.6.3.eb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+sources = [SOURCE_TAR_GZ]
+
+freetype_ver = '2.6.3'
+versionsuffix += '-freetype-%s' % freetype_ver
+dependencies = [
+    ('Python', '2.7.11'),
+    ('libjpeg-turbo', '1.4.2'),
+    ('zlib', '1.2.8'),
+    ('LibTIFF', '4.0.6'),
+    ('freetype', freetype_ver),
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/Pillow/Pillow-3.2.0-intel-2016a-Python-2.7.11-freetype-2.6.3.eb]: no libpng?

File: easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }),
+    ('psutil', '4.3.0', {
+        'source_urls': ['https://pypi.python.org/packages/source/p/psutil/'],
+    }),
+    ('simplejson', '3.8.2', {
+        'source_urls': ['https://pypi.python.org/packages/source/s/simplejson/'],
+    }),
+    # versions of SQLAlchemy & sqlalchemy-migrate are fixed to old versions by TVB, probably with good reason
+    ('SQLAlchemy', '0.7.8', {
+        'source_urls': ['https://pypi.python.org/packages/source/S/SQLAlchemy/'],
```
Comment (wpoely86) [inline easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb]: but no database in the dependencies?

**Comment by wpoely86** [comment]
what's the idea with all the bundles?

File: easybuild/easyconfigs/p/Pillow/Pillow-3.2.0-intel-2016a-Python-2.7.11-freetype-2.6.3.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/p/Pillow/Pillow-3.2.0-intel-2016a-Python-2.7.11-freetype-2.6.3.eb]: it's there as an indirect dep already, so fine?

```
$ eb Pillow-3.2.0-intel-2016a-Python-2.7.11-freetype-2.6.3.eb -D | grep libpng
 * [ ] $CFGS/l/libpng/libpng-1.6.21-intel-2016a.eb (module: libpng/1.6.21-intel-2016a)
```

**Comment by boegel** [comment]
@wpoely86 TVB is a bundle of the three actual TVB components: tvb-framework, tvb-library and tvb-data for each of which a separate easyconfig is created (since it's not a trivial Python pkg, requires `python setup.py develop` for example)

the TVB-bundle easyconfig is a collection of all Python packages that are required by TVB; some are listed as deps (because they have more deps), other as extensions to avoid that each of them needs to have a corresponding module

File: easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb [hunk 3]
```diff
@@ -0,0 +1,88 @@
+easyblock = 'Bundle'
+
+name = 'TVB-bundle'
```
Comment (boegel) [inline easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb]: @wpoely86 maybe I should rename this to `TVB-deps`?

File: easybuild/easyconfigs/p/Pillow/Pillow-3.2.0-intel-2016a-Python-2.7.11-freetype-2.6.3.eb (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyconfigs/p/Pillow/Pillow-3.2.0-intel-2016a-Python-2.7.11-freetype-2.6.3.eb]: then add it here too. Don't depend on indirect deps (else you can drop zlib too)

File: easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb (hunk 3 above)
Comment (wpoely86) [inline easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb]: good idea

File: easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb]: did you check what it needs?

Comment (boegel) [inline easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb]: Not sure how, the documentation is quite sparse...

File: easybuild/easyconfigs/p/Pillow/Pillow-3.2.0-intel-2016a-Python-2.7.11-freetype-2.6.3.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/p/Pillow/Pillow-3.2.0-intel-2016a-Python-2.7.11-freetype-2.6.3.eb]: done

File: easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb]: No mention on deps in http://docs.sqlalchemy.org/en/rel_1_0/intro.html#installation, so I guess this is good to go.

Comment (wpoely86) [inline easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb]: well, strictly speaking, it's not a dependency. But TVB will have a line somewhere like:
```python
engine = create_engine('mysql+pymysql://root:root@localhost/db')
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/0f10dadb5909c47bb2387873aa1c7725 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2457.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/e2eee5789ae6c6ede8a5f885ebb13873 for a full test report.

File: easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/t/TVB-bundle/TVB-bundle-20160618-intel-2016a-Python-2.7.11.eb]: @wpoely86 I only found stuff like

```
DB_ENGINE = create_engine(TvbProfile.current.db.DB_URL, pool_recycle=5)
```

So, it seems lik the type of DB to use is configurable, *and* the DB can be remote.

So, not a dep imho...

**Comment by boegel** [comment]
Going in, thanks for the review @wpoely86!

**Pull request merged**